cura/API/Machines.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from typing import Optional, Dict,  List, TYPE_CHECKING, Any
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot, pyqtProperty
+from UM.i18n import i18nCatalog
+from UM.Logger import Logger
+if TYPE_CHECKING:
+    from cura.CuraApplication import CuraApplication
+    from cura.PrinterOutput.PrinterOutputDevice import PrinterOutputDevice
+
+i18n_catalog = i18nCatalog("cura")
+
+##  The account API provides a version-proof bridge to use Ultimaker Accounts
+#
+#   Usage:
+#       ```
+#       from cura.API import CuraAPI
+#       api = CuraAPI()
+#       api.machines.addOutputDeviceToCurrentMachine()
+#       ```
+
+##  Since Cura doesn't have a machine class, we're going to make a fake one to make our lives a
+#   little bit easier.
+class Machine():
```

Comment:
This is not a public class, it's a private one. By making it public, it becomes a new interface (which is clearly not what we want)

Reply:
Could you please clarify what you mean by plugin specific logic and also what do you propose as an alternative? In our discussion you said Cura needs a Machine class, my hope was that this could eventually be depreciated in favor of that, but for now is just a stop-gap.